### PR TITLE
Kubernetes automation

### PR DESCRIPTION
**Pull request by saranke**
1. Provision in code to use oc/kubectl commands to get crd and events from cluster based on platform given as command line parameter.
2. Common yamls for both openshift and k8s environments.
3. Fixed failure of dedup test cases on primera arrays as dedup in not supported. Conditional pick of yaml for 3par and primera and use dedup or reduce respectively.
4. Provision to overlook access_protocol given in yaml if same is specific as command line paramter.
5. Changes in code to support all minor versions of primera, earlier it was specifically for 4.1 and 4.2.

### CODE REVIEW

File: yaml/virtual-copy/virtual-copy-base-vol-tc2.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: ci-primera3par-csp-secret
+  name: hpe3parprimera-secret-tc2
```
Comment (leenaj0hn) [inline yaml/virtual-copy/virtual-copy-base-vol-tc2.yml]: In virtual copy yaml folder, the secret name has been changed to the old name. I had modified all the secret name to match the name in Conftest.py file. Not sure why this is changed back

File: yaml/virtual-copy/virtual-copy-base-vol-tc2.yml [hunk 2]
```diff
@@ -20,24 +20,24 @@ metadata:
...
+  csi.storage.k8s.io/node-publish-secret-name: hpe3parprimera-secret-tc2
   csi.storage.k8s.io/node-publish-secret-namespace: kube-system
   # Uncomment for k8s 1.14 for resize support
   #csi.storage.k8s.io/resizer-secret-name: nimble-secret
   #csi.storage.k8s.io/resizer-secret-namespace: kube-system
   # Uncomment for k8s 1.15 for resize support
   #csi.storage.k8s.io/controller-expand-secret-name: nimble-secret
   #csi.storage.k8s.io/controller-expand-secret-namespace: kube-
-  cpg: "CI_CPG"
+  cpg: "SSD_r6"
```
Comment (leenaj0hn) [inline yaml/virtual-copy/virtual-copy-base-vol-tc2.yml]: The same I see with cpgs. These are the old file values. The CPG was changed to CI_CPG as this was common across arrays and in the default domain. This is ensure that the test runs for both primera and 3par

File: yaml/virtual-copy/virtual-copy-base-vol.yml [hunk 3]
```diff
@@ -21,24 +21,23 @@ metadata:
...
   #csi.storage.k8s.io/resizer-secret-namespace: kube-system
   # Uncomment for k8s 1.15 for resize support
   #csi.storage.k8s.io/controller-expand-secret-name: nimble-secret
   #csi.storage.k8s.io/controller-expand-secret-namespace: kube-
-  cpg: "CI_CPG"
-  snapCpg: "CI_CPG"
+  cpg: "SSD_r6"
   provisioning_type: "tpvv"
-  accessProtocol: "iscsi"
+  accessProtocol: "fc"
```
Comment (leenaj0hn) [inline yaml/virtual-copy/virtual-copy-base-vol.yml]: Same for protocol as well. These are the old values

File: yaml/virtual-copy/virtual-copy-snap-vol.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: ci-primera3par-csp-secret
+  name: hpe3parprimera-secret
```
Comment (leenaj0hn) [inline yaml/virtual-copy/virtual-copy-snap-vol.yml]: Also for virtual copy some files have been removed. eg: virtual-copy-base-vol-tc4.yml. Had added then in my last PR. Not sure why they were removed.

**Review by leenaj0hn — COMMENTED** [review]
Except for virtual copy and mutator file name changes, the rest of the changes are good.

File: yaml/virtual-copy/virtual-copy-base-vol-tc2.yml (hunk 1 above)
Comment (saranke) [reply yaml/virtual-copy/virtual-copy-base-vol-tc2.yml]: Addressed the comment and made appropriate changes. Please review.

File: yaml/virtual-copy/virtual-copy-base-vol-tc2.yml (hunk 2 above)
Comment (saranke) [reply yaml/virtual-copy/virtual-copy-base-vol-tc2.yml]: Addressed the comment and made appropriate changes. Please review.

File: yaml/virtual-copy/virtual-copy-base-vol.yml (hunk 3 above)
Comment (saranke) [reply yaml/virtual-copy/virtual-copy-base-vol.yml]: Addressed the comment and made appropriate changes. Please review.

File: yaml/virtual-copy/virtual-copy-snap-vol.yml (hunk 4 above)
Comment (saranke) [reply yaml/virtual-copy/virtual-copy-snap-vol.yml]: Addressed the comment and made appropriate changes. Please review.

**Review by leenaj0hn — APPROVED** [review]

**Pull request merged**